src/wwwroot/app/directives/candidate-vacancy-info/candidate-vacancy-info.js

```diff
@@ -122,80 +126,250 @@ function CandidateVacancyInfoController($scope, // eslint-disable-line max-state
+               each(toMultiPass, composedObjectToPass => {
+                  composedObjectToPass.vacancyStageInfos = intersectOldWithSample(composedObjectToPass,
+                  sampleStageObject.vacancyStageInfos);
+               });
+               clearToogle(canMultiPass);
+               multiPassDeffered.resolve(toMultiPass);
+            }
+         }],
+         {
+            canMultiPass,
+            toogleComposedObject: (composedObject) => {
+               composedObject.toogled = !composedObject.toogled;
+            }
+         });
+      return multiPassDeffered.promise;
+   }
+
+   function intersectOldWithSample(composedObjectToPass, sampleVacancyStageInfos) {
+      let newVSIs = map(getVSIsToPass(composedObjectToPass, sampleVacancyStageInfos), vsi => {
+         let newVsi = Object.assign({}, vsi);
+         newVsi = Object.assign(newVsi, {
+            vacancyId: composedObjectToPass.vacancyId
+         });
+         delete newVsi.id;
+         return newVsi;
+      });
+      let combinedVSIs = intersectOldAndNewVsis(composedObjectToPass, newVSIs);
+      if (!isThereActiveStage(combinedVSIs)) {
+         combinedVSIs.push(getActiveStage(combinedVSIs, composedObjectToPass));
+      }
+      return combinedVSIs;
+   }
+   function intersectOldAndNewVsis(composedObject, newVSIs) {
+      let newAndOld = [...composedObject.vacancyStageInfos, ...newVSIs];
+      let uniqStages = uniq(map(newAndOld, vsi => {
+         return vsi.stageId;
+      }));
+      let intersection = map(uniqStages, stageId => {
+         let newVSI = find(newVSIs, ['stageId', stageId]);
+         let oldVSI = find(composedObject.vacancyStageInfos, ['stageId', stageId]);
+         if (!newVSI) {
+            return oldVSI;
+         }
+         if (oldVSI) {
+            return Object.assign(Object.assign({}, oldVSI), {
+               vacancyId: composedObject.vacancyId,
+               stageState: newVSI.stageState,
+               comment: newVSI.comment
+            });
+         }
+         let newObjectOfnewVSI = Object.assign(Object.assign({},newVSI), {
+            vacancyId: composedObject.vacancyId
+         });
+         delete newObjectOfnewVSI.id;
+         return newObjectOfnewVSI;
+      });
+      return intersection;
+   }
+   function getObjectsToPass(objectThatCanBeMultiPassed) {
+      return filter(objectThatCanBeMultiPassed, x => {
+         return x.toogled;
+      });
+   }
+   function clearToogle(objectThatCanBeMultiPassed) {
+      each(objectThatCanBeMultiPassed, x => delete x.toogled);
+   }
+   function isThereActiveStage(combinedVSIs) {
+      return find(combinedVSIs, ['stageState', STAGE_STATES.Active]);
+   }
+   function getActiveStage(combinedVSIs, composedObject) {
+      let latestStageId = reduce(combinedVSIs, (max, vsi) => {
+         if (max < vsi.stageId && vsi.stageState === STAGE_STATES.Passed) {
+            return vsi.stageId;
+         }
+         return max;
```

Comment:
can be ternar:
return max < vsi.stageId && vsi.stageState === STAGE_STATES.Passed) ?
            vsi.stageId :
            max;